src/commands.cpp

```diff
@@ -0,0 +1,152 @@
+    }
+
+    // Check keys and value types
+    for (const auto &[key, json_definition] : definition)
+    {
+        const auto search = json_obj.find(key); // Current value that will be checked
+
+        if (search == json_obj.end())
+        {
+            if (json_definition.required) // Check that key exists if required
+                throw std::runtime_error("CommandDefinitions::parse_check_json: Missing required key: " + key);
+            else // If key is not required and doesn't exist, continue with next key
+                continue;
+        }
+
+        // Check if value is in bound (only supported for numbers)
+        if (!json_definition.check_bounds(search))
+        {
+            std::string error_msg = "CommandDefinitions::parse_check_json: [Key: '" + key + "'] Value '" + search->dump() + "' is out of bounds: ";
+
+            if (json_definition.min_val.has_value() && json_definition.max_val.has_value()) // Message when min and max values are defined
+                error_msg += "[" + std::to_string(json_definition.min_val.value()) + "; " + std::to_string(json_definition.max_val.value()) + "]";
+            else if (json_definition.min_val.has_value()) // Message when only the min value is defined
+                error_msg += "> " + std::to_string(json_definition.min_val.value());
+            else if (json_definition.max_val.has_value()) // Message when only the max value is defined
+                error_msg += "< " + std::to_string(json_definition.max_val.value());
+
+            throw std::runtime_error(error_msg);
+        }
+
+        // Check that type of value matches the definition
+        bool type_check = false;
+        for (const auto &data_type : json_definition.data_types) // Loop through all the allowed data types
+        {
+            switch (data_type)
+            {
+            case null:
+                if (search->is_null())
+                    type_check = true;
+                break;
+            case boolean:
+                if (search->is_boolean())
+                    type_check = true;
+                break;
+            case number:
+                if (search->is_number())
+                    type_check = true;
+                break;
+            case number_float:
+                if (search->is_number_float())
+                    type_check = true;
+                break;
+            case number_integer:
+                if (search->is_number_integer())
+                    type_check = true;
+                break;
+            case number_unsigned:
+                if (search->is_number_unsigned())
+                    type_check = true;
+                break;
+            default:
+                throw std::runtime_error("CommandDefinitions::parse_check_json: Unknown data_type provided: " + data_type);
+            }
+
+            if (type_check)
+                break;
+        }
+
+        if (!type_check) // Type check failed
```

Comment:
Würde ich in die json_definition Klasse ausgliedern.

Reply:
Da diese Funktion nur hier gebraucht wird, halte ich eine Auslagerung für unnötig und macht den Code eher schlechter nachvollziehbar.